rskj-core/src/main/java/co/rsk/net/discovery/message/PeerDiscoveryMessage.java

```diff
@@ -175,4 +182,15 @@ public String toString() {
                 .append("type", ByteUtil.toHexString(type))
                 .append("data", ByteUtil.toHexString(data)).toString();
     }
+
+    protected String extractMessageId(RLPItem chk) {
+        String rawMessageId = new String(chk.getRLPData(), StandardCharsets.UTF_8);
+        try {
+            return UUID.fromString(rawMessageId).toString();
```

Comment:
Probably worth it to check that the version of the UUID is 4 while we are at it :)

Reply:
Suggestion applied!